src/Cafe/HW/Latte/Renderer/Renderer.cpp

```diff
@@ -101,60 +102,86 @@ fs::path _GenerateScreenshotFilename(bool isDRC)
 			screenshotPath.append(fmt::format("{}.png", screenshotFileName));
 		else
 			screenshotPath.append(fmt::format("{}_{}.png", screenshotFileName, i + 1));
+		
 		std::error_code ec;
-		if (!fs::exists(screenshotPath))
+		bool exists = fs::exists(screenshotPath, ec);
+		
+		if (!ec && !exists)
 			return screenshotPath;
 	}
-	return screenshotPath; // if all exist checks fail, return the last path we tried
+	return std::nullopt;
 }
 
 std::mutex s_clipboardMutex;
 
-void Renderer::SaveScreenshot(const std::vector<uint8>& rgb_data, int width, int height, bool mainWindow) const
+static bool SaveScreenshotToClipboard(const wxImage &image)
+{
+	bool success = false;
+
+	s_clipboardMutex.lock();
+	if (wxTheClipboard->Open())
+	{
+		wxTheClipboard->SetData(new wxImageDataObject(image));
+		wxTheClipboard->Close();
+		success = true;
+	}
+	s_clipboardMutex.unlock();
+
+	return success;
+}
+
+static bool SaveScreenshotToFile(const wxImage &image, bool mainWindow)
+{
+	auto path = _GenerateScreenshotFilename(!mainWindow);
+	if (!path) return false;
+
+	std::error_code ec;
+	fs::create_directories(path->parent_path(), ec);
+	if (ec) return false;
+
+	wxLogNull _logNo;
```

Comment:
Could you add a comment on why suppressing the log is necessary?

Reply:
Sure. Since a `wxImage` had to be constructed to set the clipboard data, I don't think it's too egregious to use `wxImage::SaveFile`, despite having to suppress the logs here. Without it, a redundant message box will pop up in certain cases (e.g. no write permissions on screenshots dir.)